Nprpr 22 и Nprp 21 две модалки "редактировать финансовую операцию" и "удалить финансовую операцию"

sleipnir22 opened the pull request:
От тебя добавил RedirectToAction на Details проекта

sleipnir22 committed:
Сделал модалку удаления категории и набросал редактирование

sleipnir22 committed:
Замерджил мастер со своей веткой. Да простит мне это господь бог наш иисус христос

sleipnir22 committed:
Пофиксил разные баги + сделал перенаправление после удаления или редактирования категории

yudakov1814 commented on FamilyBudget/Views/Home/Details.cshtml:
```diff
@@ -178,6 +185,8 @@
                         }
                     </td>
                     <td class="col-3">@fo.FinType</td>
+                    <td id="@fo.Id" class="bi bi-pencil finOperationEdit"></td>
+                    <td id="@fo.Id" class="bi bi-x-lg finOperationDelete"></td>
```
Нужно добавить style="cursor: pointer", иначе не понятно, что иконки кликабельны

yudakov1814 commented on FamilyBudget/Views/Home/Details.cshtml:
```diff
@@ -5,6 +5,13 @@
+    <div id="finOperationDeleteDialogContent" class="modal-dialog modal-dialog-centered modal-lg"></div>
+</div>
+
+<div id="finOperationEditModal" class="modal fade">
+    <div id="finOperationEditDialogContent" class="modal-dialog modal-dialog-centered"></div>
+</div>
```
Не так просто. Для доходов и для расходов модалка должна отличаться. Посмотри как это сделано в addFinOpModal. Там есть AddCharge.cshtml и AddIncome.cshtml

yudakov1814 replied:
И вообще посмотри внимательно как создается новый расход. Там есть фейковый ProjectMember "Семья". При редактировании ты его должен поддержать. Еще глянь в FinOperationsController метод Create, там есть проверка if (finOperation.ProjectMemberId == -1). Тебе тоже самое нужно в edit добавить

sleipnir22 committed:
Добавил изменение курсора при наведении

sleipnir22 committed:
1)вместо двух модалок оставил одну, но при условном отображении категории расхода (то есть элемент отображается, если FinType==FinType.Charge; в ином же случае не отображается);
2) поправил остальные недочеты + сделал ввод категории привел к тому виду, в котором он в остальных модалках

sleipnir22 committed:
Фикс. Происходило неявное преобразование int? к int

sleipnir22 committed:
.

sleipnir22 committed:
Revert "."

This reverts commit 8ca4eee58fde73e90b31e111746c2005e0ba3c61.

sleipnir22 committed:
Revert "Фикс. Происходило неявное преобразование int? к int"

This reverts commit c5f5261cc8bcd0b8cd4356db9dd0b2ef60e3b2db.

sleipnir22 committed:
проблема с преобразованием int? к int

sleipnir22 committed:
Merge branch 'master' into NPRPR-22

sleipnir22 committed:
долгожданный фикс

sleipnir22 committed:
Фикс редактирования финансовых операций

sleipnir22 committed:
вроде лишнее спрятал
